pkg/api/mlflow/service/artifact/service.go

```diff
@@ -85,10 +87,20 @@ func (s Service) GetArtifact(
 		run.ArtifactURI, req.Path,
 	)
 	if err != nil {
-		return nil, api.NewInternalError(
+		msg := fmt.Sprintf(
 			"error getting artifact object for URI: %s",
 			filepath.Join(run.ArtifactURI, req.Path),
 		)
+		statError := regexp.MustCompile(storage.PathError)
+		isDirError := regexp.MustCompile(storage.IsDirError)
+		switch {
+		case statError.MatchString(err.Error()):
+			return nil, api.NewResourceDoesNotExistError(msg)
+		case isDirError.MatchString(err.Error()):
+			return nil, api.NewInvalidParameterValueError(msg)
+		default:
+			return nil, api.NewInternalError(msg)
+		}
```

Comment:
why do you use this weird regex matching instead of the standard `errors.Is`, or even `eris.Is`?

Reply:
also, if `os.Stat` or `os.Open` failed with a different error than `fs.ErrNotExist`, it _is_ an internal error, not a `ResourceDoesNotExistError`!

Reply:
I can switch to eris.Is()